sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/TransientIOErrorsRetryingIterator.scala

```diff
@@ -78,7 +108,6 @@ private class TransientIOErrorsRetryingIterator
         val retryIntervalInMs = rnd.nextInt(maxRetryIntervalInMs)
 
         try {
-          read.lock()
```

Comment:
Why are we removing the read lock ?

Reply:
Iterator in itself is inheritently not thread-safe - the calling pattern enforces hasNext followed by next to happen. Locks were only there because with handle concurrency with calls into hasNext/next could have been concurrent